build warnings

clang finds a bit more

```
/vagrant/src/ast/int_parser.cpp:13:33: warning: unused parameter 'num' [-Wunused-parameter]
T _parse_int(const std::string &num, size_t *idx, int base)
                                ^
/vagrant/src/ast/int_parser.cpp:13:46: warning: unused parameter 'idx' [-Wunused-parameter]
T _parse_int(const std::string &num, size_t *idx, int base)
                                             ^
/vagrant/src/ast/int_parser.cpp:13:55: warning: unused parameter 'base' [-Wunused-parameter]
T _parse_int(const std::string &num, size_t *idx, int base)
```
```
/vagrant/src/ast/resource_analyser.cpp:16:3: warning: loop will run at most once (loop increment never executed) [-Wunreachable-code-loop-increment]
  for (auto &attach_point : *probe->attach_points)
  ^~~
```
```
/vagrant/src/attached_probe.cpp:113:10: warning: comparison of different enumeration types in switch statement ('bpf_prog_type' and 'libbpf::bpf_prog_type') [-Wenum-compare-switch]
    case libbpf::BPF_PROG_TYPE_TRACING:    return "BPF_PROG_TYPE_TRACING";    break;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/vagrant/src/attached_probe.cpp:112:10: warning: comparison of different enumeration types in switch statement ('bpf_prog_type' and 'libbpf::bpf_prog_type') [-Wenum-compare-switch]
    case libbpf::BPF_PROG_TYPE_PERF_EVENT: return "BPF_PROG_TYPE_PERF_EVENT"; break;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/vagrant/src/attached_probe.cpp:111:10: warning: comparison of different enumeration types in switch statement ('bpf_prog_type' and 'libbpf::bpf_prog_type') [-Wenum-compare-switch]
    case libbpf::BPF_PROG_TYPE_TRACEPOINT: return "BPF_PROG_TYPE_TRACEPOINT"; break;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/vagrant/src/attached_probe.cpp:110:10: warning: comparison of different enumeration types in switch statement ('bpf_prog_type' and 'libbpf::bpf_prog_type') [-Wenum-compare-switch]
    case libbpf::BPF_PROG_TYPE_KPROBE:     return "BPF_PROG_TYPE_KPROBE";     break;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
4 warnings generated.
```

```
/vagrant/src/main.cpp:861:16: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    bytecode = std::move(bpforc->getBytecode());
               ^
/vagrant/src/main.cpp:861:16: note: remove std::move call here
    bytecode = std::move(bpforc->getBytecode());
               ^~~~~~~~~~
```

--- a/src/ast/int_parser.cpp
+++ b/src/ast/int_parser.cpp
@@ -10,7 +10,9 @@
 namespace {
 
 template <typename T>
-T _parse_int(const std::string &num, size_t *idx, int base)
+T _parse_int(const std::string &num __attribute__((unused)),
+             size_t *idx __attribute__((unused)),
+             int base __attribute__((unused)))
 {
   static_assert(not std::is_same_v<T, T>,
                 "BUG: _parse_int not implemented for type");

--- a/src/ast/resource_analyser.cpp
+++ b/src/ast/resource_analyser.cpp
@@ -13,8 +13,10 @@ namespace {
 // well formed.
 ProbeType single_provider_type_postsema(Probe *probe)
 {
-  for (auto &attach_point : *probe->attach_points)
-    return probetype(attach_point->provider);
+  if (!probe->attach_points->empty())
+  {
+    return probetype(probe->attach_points->at(0)->provider);
+  }
 
   return ProbeType::invalid;
 }

--- a/src/attached_probe.cpp
+++ b/src/attached_probe.cpp
@@ -104,7 +104,7 @@ bpf_prog_type progtype(ProbeType t)
 
 std::string progtypeName(bpf_prog_type t)
 {
-  switch (t)
+  switch (static_cast<libbpf::bpf_prog_type>(t))
   {
     // clang-format off
     case libbpf::BPF_PROG_TYPE_KPROBE:     return "BPF_PROG_TYPE_KPROBE";     break;

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -858,7 +858,7 @@ int main(int argc, char* argv[])
       bpforc->dump(os);
     }
 
-    bytecode = std::move(bpforc->getBytecode());
+    bytecode = bpforc->getBytecode();
   }
   catch (const std::system_error& ex)
   {
@@ -902,7 +902,7 @@ int main(int argc, char* argv[])
   else if (!bt_quiet)
     bpftrace.out_->attached_probes(num_probes);
 
-  err = bpftrace.run(std::move(bytecode));
+  err = bpftrace.run(bytecode);
   if (err)
     return err;
 

